src/goMain.ts

```diff
@@ -87,14 +87,24 @@ export function activate(ctx: vscode.ExtensionContext): void {
 		ctx.globalState.update('goroot', currentGoroot);
 
 		offerToInstallTools();
-		let langServerAvailable = checkLanguageServer();
-		if (langServerAvailable) {
-			let langServerFlags: string[] = vscode.workspace.getConfiguration('go')['languageServerFlags'] || [];
+		let supportedLangServerFlags = checkLanguageServer();
+		if (supportedLangServerFlags) {
+			let configuredLangServerFlags: string[] = vscode.workspace.getConfiguration('go')['languageServerFlags'] || [];
+			configuredLangServerFlags = [
+				...configuredLangServerFlags,
+				'-gocodecompletion',
+				`-func-snippet-enabled=${vscode.workspace.getConfiguration('go')['useCodeSnippetsOnFunctionSuggest']}`
+			];
+
+			let applicableFlags = configuredLangServerFlags.filter(f => {
+				return supportedLangServerFlags.some(supported => f.startsWith(supported));
```

Comment:
why `startsWith` and not `===` ?

Reply:
Because for example the supported flags will contain `-func-snippet-enabled` but we also pass a value like `-func-snippet-enabled=false`.

Reply:
Ah! Makes sense